### PR TITLE
Ed25519 test only natives

### PR DESCRIPTION
**Pull request by zjma**
### Description

This PR adds support for test-only natives in Aptos VM via a feature flag. There was a previous attempt of doing this via a boolean flag that we moved away from in #5129.

This PR also adds test-only native functions for Ed25519 signature generation, so we can get rid of hardcoded public key/signatures in our Move module tests!

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/5272)

### CODE REVIEW

**Comment by zjma** [comment]
previous attempt: https://github.com/aptos-labs/aptos-core/pull/5129/files

**Review by wrwg — COMMENTED** [review]
Handling of test-only natives looks good. I leave the core of the logic to review for Alin.

File: aptos-move/framework/src/natives/cryptography/ed25519.rs [hunk 1]
```diff
@@ -136,17 +138,59 @@ pub struct GasParameters {
...
-            "public_key_validate_internal",
-            make_native_from_func(gas_params.clone(), native_public_key_validate),
-        ),
-        (
-            "signature_verify_strict_internal",
-            make_native_from_func(gas_params, native_signature_verify_strict),
-        ),
-    ];
-
+    let mut natives = vec![];
```
Comment (alinush) [inline aptos-move/framework/src/natives/cryptography/ed25519.rs:141]: You could've kept the `Vec` from above, no? It's cleaner/easier to red.

Then, you could similarly declare a `test_natives` `Vec` and use [`append`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.append) to append it to `natives` based on the feature flag.

**Review by alinush — APPROVED** [review]

**Comment by alinush** [comment]
Core logic looks good.

**Review by vgao1996 — APPROVED** [review]

**Comment by zjma** [comment]
@chloeqjz @CapCap @areshand @movekevin check out this new way to generate ed25519 signatures in tests!
Support for BLS/MultiEd25519 signatures are also coming soon.

**Pull request merged**